pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,412 @@
+	resultCh chan []jobrunaggregatorapi.JobRunInfo, jobName string, jobRunLocator jobrunaggregatorlib.JobRunLocator) {
+	defer wg.Done()
+	errorsInARow := 0
+	for {
+		jobRuns, err := jobRunLocator.FindRelatedJobs(ctx)
+		if err != nil {
+			if errorsInARow > 20 {
+				fmt.Printf("give up finding job runs for %s after retries: %v", jobName, err)
+				return
+			}
+			errorsInARow++
+			fmt.Printf("error finding job runs for %s: %v", jobName, err)
+		} else {
+			resultCh <- jobRuns
+			return
+		}
+
+		fmt.Printf("   waiting and will attempt to find related jobs for %s in a minute\n", jobName)
+		select {
+		case <-ctx.Done():
+			// Simply return. Caller will check ctx and return error
+			return
+		case <-time.After(1 * time.Minute):
+			continue
+		}
+	}
+}
+
+// GetRelatedJobRuns gets all related job runs for analysis
+func (o *JobRunTestCaseAnalyzerOptions) GetRelatedJobRuns(ctx context.Context) ([]jobrunaggregatorapi.JobRunInfo, error) {
+	var jobRunsToReturn []jobrunaggregatorapi.JobRunInfo
+	jobs, err := o.getRelatedJobs(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get related jobs: %w", err)
+	}
+
+	waitGroup := sync.WaitGroup{}
+	resultCh := make(chan []jobrunaggregatorapi.JobRunInfo, len(jobs))
+	for i := range jobs {
+		job := jobs[i]
+		var jobRunLocator jobrunaggregatorlib.JobRunLocator
+
+		if len(o.payloadTag) > 0 {
+			jobRunLocator = jobrunaggregatorlib.NewPayloadAnalysisJobLocatorForReleaseController(
+				job.JobName,
+				o.payloadTag,
+				o.jobRunStartEstimate,
+				o.ciDataClient,
+				o.ciGCSClient,
+				o.gcsClient,
+				"origin-ci-test",
+			)
+		}
+		if len(o.payloadInvocationID) > 0 {
+			jobRunLocator = jobrunaggregatorlib.NewPayloadAnalysisJobLocatorForPR(
+				job.JobName,
+				o.payloadInvocationID,
+				jobrunaggregatorlib.PayloadInvocationIDLabel,
+				o.jobRunStartEstimate,
+				o.ciDataClient,
+				o.ciGCSClient,
+				o.gcsClient,
+				"origin-ci-test",
+				(*o.jobGCSPrefixes)[i].gcsPrefix,
+			)
+		}
+
+		fmt.Printf("  launching findJobRunsWithRetry for %q\n", job.JobName)
+
+		waitGroup.Add(1)
+
+		go o.findJobRunsWithRetry(ctx, &waitGroup, resultCh, job.JobName, jobRunLocator)
+	}
+	waitGroup.Wait()
+	close(resultCh)
+
+	// drain the result channel first
+	for jobRuns := range resultCh {
+		jobRunsToReturn = append(jobRunsToReturn, jobRuns...)
+	}
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	default:
+		break
+	}
+	return jobRunsToReturn, nil
+}
+
+func (o *JobRunTestCaseAnalyzerOptions) runTestCaseCheckers(ctx context.Context,
+	finishedJobRuns []jobrunaggregatorapi.JobRunInfo, unfinishedJobRuns []jobrunaggregatorapi.JobRunInfo) *junit.TestSuite {
+	suiteName := "test case checker"
+	if len(o.platform) != 0 {
+		suiteName += fmt.Sprintf(" for platform %s", o.platform)
+	}
+	if len(o.network) != 0 {
+		suiteName += fmt.Sprintf(" with network %s", o.network)
+	}
+	if len(o.infrastructure) != 0 {
+		suiteName += fmt.Sprintf(" and infrastructure %s", o.infrastructure)
+	}
```

Comment:
This feels like too much granularity leaking into suite names, I'd expect fewer test suites with many tests within, in this case it looks like we're going to be close to 1-1 suite-test relationship. Could we move this info into the test name instead and use the suite for a generic bucket for all of these types of tests? This will give us better insight into how a test is behaving on each platform in Sippy, which is also not well setup for dynamic suite names.

Reply:
This is where I need some suggestions. 

This is the current format:

<testsuite name="test case checker for platform aws with network ovn and infrastructure ipi" tests="1" skipped="0" failures="1" time="0">
    <testcase name="test 'cluster install/install should succeed: overall/platform: aws network: ovn infrastructure ipi' should have required number of successful passes across jobRuns" time="0">
    </testcase>
</testsuite>

I am thinking of changing this to:
<testsuite name="payload-cross-job" tests="1" skipped="0" failures="1" time="0">
    <testcase name="test 'cluster install/install should succeed: overall/platform: aws network: ovn infrastructure ipi' should have required number of successful passes across jobRuns" time="0">
    </testcase>
</testsuite>

Any improvement on the test name?